server/config/index.ts

```diff
@@ -0,0 +1,16 @@
+export type Config = {
+  regenerateOnUpdate: boolean;
+};
+
+const DEFAULT_CONFIG = {
+  regenerateOnUpdate: false,
+} as const satisfies Config;
+
+export default {
+  default: () => DEFAULT_CONFIG,
+  validator: (config: any) => {
```

Comment:
I'm not sure about the correct way to type this validation function... I just left `any` and no type guard as return.